src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -2063,13 +2063,16 @@ public StringBuilder Replace(char oldChar, char newChar, int startIndex, int cou
                 {
                     int curInChunk = Math.Max(startIndexInChunk, 0);
                     int endInChunk = Math.Min(chunk.m_ChunkLength, endIndexInChunk);
-                    while (curInChunk < endInChunk)
+
+                    Span<char> span = chunk.m_ChunkChars.AsSpan(curInChunk, endInChunk - curInChunk);
+                    int i;
+                    while ((i = span.IndexOf(oldChar)) >= 0)
```

Comment:
Should we propose `public static int Replace<T>(this Span<T> span, T value) where T : IEquatable<T>` in MemoryExtensions?

Reply:
Open an issue :)

Reply:
Details..

Reply:
https://github.com/dotnet/runtime/issues/29758
also discussion in https://github.com/dotnet/runtime/issues/22434
The tricky part seems to be that in the general case, the output may change length..